### PR TITLE
SSHFP for 2018

### PR DESCRIPTION
**Pull request by ph448**
This PR intends to add sending Ed25519 host keys and remove sending SHA1 hashes to Jackdaw.
Tests could have been updated to test for these, but as we mock them and not use the full suite of host keys anyway I saw little value.
While madly grepping the source tree I also came accross a typo where Ed25516 was used instead.
Fixes uisautomation/mws-panel#18.
The PR code has not yet been tested, so please only review, do not merge.

### CODE REVIEW

File: mws/apimws/xen.py [hunk 1]
```diff
@@ -78,25 +78,18 @@ def secrets_prealocation_vm(vm):
...
-                    set_sshfp(service.site.test_service.network_configuration.name, SiteKey.ALGORITHMS[keytype],
-                              SiteKey.FP_TYPES[fptype], fp)
-                    set_sshfp(vm.network_configuration.name, SiteKey.ALGORITHMS[keytype], SiteKey.FP_TYPES[fptype],
-                              fp)
-                except Exception as e:
-                    LOGGER.error("Error while trying to set up sshfp records. \nkeytype: %s\nfptype: %s\nexception: %s"
-                                 % (keytype, fptype, str(e.__class__)+" "+str(e)))
-                    pass
+        try:
+            fptype == 'SHA256'
```
Comment (abrahammartin) [inline mws/apimws/xen.py]: fptype is not set

File: mws/apimws/xen.py [hunk 2]
```diff
@@ -78,25 +78,18 @@ def secrets_prealocation_vm(vm):
...
+            set_sshfp(service.network_configuration.name, SiteKey.ALGORITHMS[keytype],
```
Comment (abrahammartin) [inline mws/apimws/xen.py]: fp is not set

File: mws/sitesmanagement/signals.py [hunk 3]
```diff
@@ -18,15 +18,14 @@ def check_main_domain_name(instance, **kwargs):
 @receiver(pre_delete, sender=SiteKey)
 def delete_sshfp_from_dns(instance, **kwargs):
     '''Delete SSHFP records from the DNS using the DNS API when a SiteKey is deleted from the database'''
-    if instance.type != "ED25519":
-        for service in instance.site.services.all():
+    for service in instance.site.services.all():
+        for fptype in SiteKey.FP_TYPES:
+            delete_sshfp(service.network_configuration.name, SiteKey.ALGORITHMS[instance.type],
```
Comment (abrahammartin) [inline mws/sitesmanagement/signals.py]: this will break trying to delete SHA1 entries that no longer exists in the DNS

Comment (ph448) [reply mws/sitesmanagement/signals.py]: I wasn't sure about what the ipreg api does with such requests. back to the drawing board on this one...

File: mws/apimws/ipreg.py [hunk 4]
```diff
@@ -65,9 +66,17 @@ def delete_cname(hostname):
 
 
 def find_sshfp(hostname):
+    result = []
```
Comment (abrahammartin) [inline mws/apimws/ipreg.py]: Code looks ok but I don't think it provides the same functionality as the code it replaces. The old code checked in the ip_reg database if an sshfp entry existed for the given hostname while the new code checks if the DNS servers have a sshfp entry for the given hostname. The two are different, and before the DNS gets updated (at 53 mins every hour) the former may respond true, and the latter may respond false.

Comment (abrahammartin) [reply mws/apimws/ipreg.py]: For documentation on how the find_sshfp ipreg_api works: https://github.com/uisautomation/mws-misc/blob/f92592afe6ff51423b66119a0cfa0a8c3354e837/dns/API#L142

and the implementation is in

https://github.com/uisautomation/mws-misc/blob/f92592afe6ff51423b66119a0cfa0a8c3354e837/dns/util/perl/ipreg_api.pl#L292

File: mws/requirements.txt [hunk 5]
```diff
@@ -17,3 +17,4 @@ redis
 requests
 python-dateutil
 ansible<2.5
+dnspython
```
Comment (abrahammartin) [inline mws/requirements.txt]: I think you don't need this requirement anymore

**Pull request merged**